src/forum/controller.rs

```diff
@@ -106,6 +105,23 @@ pub struct CreateReply {
     pub message: String,
 }
 
+pub async fn handle_delete_post(
+    Path(post_id): Path<i32>,
+    State(app_state): State<AppState>,
+) -> ForumResult<Response> {
+    let conn = get_connection(&app_state)?;
+
+    ForumPost::soft_delete_post(&conn, post_id as usize)?;
```

Comment:
Why not align the type of post_id? So just one numeric type.

It could be u64 for now, unless there's a reason it is i32?

Reply:
Good point.